### PR TITLE
BG-11648: Add `fromPrivateKeyBuffer` from provajs-lib

### PR DESCRIPTION
**Pull request by tylerlevine**
https://bitgoinc.atlassian.net/browse/BG-11648

### CODE REVIEW

File: test/ecpair.js [hunk 1]
```diff
@@ -299,4 +299,18 @@ describe('ECPair', function () {
...
+  describe('fromPrivateKeyBuffer', function () {
+    var ecPair
+
+    before(function () {
+      ecPair = ECPair.makeRandom()
+    })
+
+    it('reconstructs an ECPair from the private key buffer', function () {
+      var prvKeyBuffer = ecPair.getPrivateKeyBuffer()
+      var reconstructedEcPair = ECPair.fromPrivateKeyBuffer(prvKeyBuffer)
```
Comment (argjv) [inline test/ecpair.js]: Add a test using different networks

**Comment by argjv** [comment]
Do we have to add RMG to the network objects? Or are its params the same as Bitcoin?

File: src/ecpair.js [hunk 2]
```diff
@@ -59,6 +60,26 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
   })
 }
 
+ECPair.fromPrivateKeyBuffer = function (buffer, network) {
+  try {
+    typeforce(typeforce.BufferN(32), buffer)
+  } catch (e) {
+    throw new Error(e.message)
```
Comment (tylerlevine) [inline src/ecpair.js]: this is dumb

File: src/ecpair.js [hunk 3]
```diff
@@ -59,6 +60,26 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
...
+  }
+  var d = BigInteger.fromBuffer(buffer)
+
+  if (d.signum() <= 0 || d.compareTo(curve.n) >= 0) {
+    throw new Error('bad private key buffer')
```
Comment (tylerlevine) [inline src/ecpair.js]: test this case

File: test/ecpair.js (hunk 1 above)
Comment (tylerlevine) [reply test/ecpair.js]: I'm not sure what that would accomplish. `fromPrivateKeyBuffer`'s behavior does not depend on the network.

Comment (argjv) [reply test/ecpair.js]: Ah true, nvm.

File: src/ecpair.js [hunk 4]
```diff
@@ -59,6 +60,25 @@ ECPair.fromPublicKeyBuffer = function (buffer, network) {
   })
 }
 
+ECPair.fromPrivateKeyBuffer = function (buffer, network) {
```
Comment (argjv) [inline src/ecpair.js]: Can you add method description?

Comment (tylerlevine) [reply src/ecpair.js]: done

**Pull request merged**